File: config_server/service/common/http_response.go
// Copyright 2022 iLogtail Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package common

type httpStatus struct {
	Status int
	Code   string
}

var (
	Accept                 = httpStatus{200, "Accept"}
	BadRequest             = httpStatus{400, "BadRequest"}
	ConfigAlreadyExist     = httpStatus{400, "ConfigAlreadyExist"}
	ConfigNotExist         = httpStatus{404, "ConfigNotExist"}
	InternalServerError    = httpStatus{500, "InternalServerError"}
	InvalidParameter       = httpStatus{400, "InvalidParameter"}
	AgentAlreadyExist      = httpStatus{400, "AgentAlreadyExist"}
	AgentGroupAlreadyExist = httpStatus{400, "AgentGroupAlreadyExist"}
	AgentGroupNotExist     = httpStatus{404, "AgentGroupNotExist"}
	AgentNotExist          = httpStatus{404, "AgentNotExist"}
	RequestTimeout         = httpStatus{500, "RequestTimeout"}
	ServerBusy             = httpStatus{503, "ServerBusy"}
)

File: core/common/http/HttpResponse.cpp
// Copyright 2024 iLogtail Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include "common/http/HttpResponse.h"

using namespace std;

namespace logtail {

bool caseInsensitiveComp(const char lhs, const char rhs) {
    return tolower(lhs) < tolower(rhs);
}

bool compareHeader(const std::string& lhs, const std::string& rhs) {
    return lexicographical_compare(lhs.begin(), lhs.end(), rhs.begin(), rhs.end(), caseInsensitiveComp);
}

size_t DefaultWriteCallback(char* buffer, size_t size, size_t nmemb, void* data) {
    unsigned long sizes = size * nmemb;

    if (buffer == NULL) {
        return 0;
    }

    static_cast<string*>(data)->append(buffer, sizes);
    return sizes;
}

} // namespace logtail
